src/value/mod.rs

```diff
@@ -258,10 +259,13 @@ impl Eq for AnyInt {}
 
 /// Represents a either a borrowed or owned Ion datum.  There are/will be specific APIs for
 /// _borrowed_ and _owned_ implementations, but this trait unifies operations on either.
-pub trait Element {
-    type SymbolToken: SymbolToken + ?Sized;
-    type Sequence: Sequence<Element = Self> + ?Sized;
-    type Struct: Struct<FieldName = Self::SymbolToken, Element = Self> + ?Sized;
+pub trait Element
+where
+    Self: From<i64> + From<bool> + From<Decimal> + From<Timestamp> + From<f64> + From<BigInt>,
+{
+    type SymbolToken: SymbolToken + ?Sized + Debug + PartialEq;
+    type Sequence: Sequence<Element = Self> + ?Sized + Debug + PartialEq;
+    type Struct: Struct<FieldName = Self::SymbolToken, Element = Self> + ?Sized + Debug + PartialEq;
```

Comment:
Since you added `Debug` to all of the downstream traits, should `Element` have `Self: Debug`?